oioioi/contests/admin.py

```diff
@@ -848,6 +850,34 @@ def score_display(self, instance):
     score_display.short_description = _("Score")
     score_display.admin_order_field = 'score_with_nulls_smallest'
 
+    def score_diff_display(self, instance):
+        contest_controller = instance.problem_instance.contest.controller
+        pi_controller = instance.problem_instance.controller
+        if not contest_controller.display_score_change() or instance.kind != 'NORMAL':
+            return format_html('<span class="text-secondary">-</span>')
+
+        try:
+            previous_submission = pi_controller.get_last_scored_submission(
+                instance.user,
+                instance.problem_instance,
+                before=instance.date,
+            )
+        except Submission.DoesNotExist:
+            previous_submission = None
+        try:
+            curr_submission = pi_controller.get_last_scored_submission(
+                instance.user,
+                instance.problem_instance,
+                before=instance.date,
+                include_current=True,
+            )
+        except Submission.DoesNotExist:
+            curr_submission = None
```

Comment:
doesn't this always equal `instance`?

Reply:
No, for example when the contests takes the highest scored submission as the score for a problem

Reply:
Right. But then in my opinion `_calculate_score_change` should adhere to contest rules, so an override in MPContestController for instance should return:
`return cls(max(after.value, before.value))`